### ISSUE TITLE
Formspecs: Textlist selected id does not work for first textlist in formspec

### ISSUE THREAD

**Issue by Thomas--S** [minetest/minetest#7141]
##### Issue type

 - Bug report
 
##### Minetest version

c34d28bad6d3106d036570cb40ed3dddc6b4845c

Also tested in 0.4.16, the bug exists there, too.

##### OS / Hardware

Operating system: Debian

##### Summary

https://github.com/minetest/minetest/blob/0358ae789a6aa5c161c3b9017651b7d2c2f28fd8/doc/lua_api.txt#L2166

The `<selected idx>` field does not work for the first textlist in a formspec.
For the first textlist, the last entry is selected.

It seems to work correctly for node formspecs, only `minetest.show_formspec` formspecs are affected.

##### Steps to reproduce

```lua
minetest.register_craftitem(":test:test", {
	description = "Test Textlist Bug",
	inventory_image = "default_cobble.png^default_stick.png",
	on_use = function(itemstack, user, pointed_thing)
		if not user then
			return nil
		end
		local playername = user:get_player_name()
		local fs = "size[12,9]"
		fs = fs .. "textlist[0,1;3,7;test1;a,b,c,d,e,f;4;false]"
		fs = fs .. "textlist[4,1;3,7;test2;a,b,c,d,e,f;4;false]"
		fs = fs .. "textlist[8,1;3,7;test3;a,b,c,d,e,f;4;false]"
		fs = fs .. "label[1,8;The selected id field does not work for the first textlist in a formspec.]"
		minetest.show_formspec(playername, "test:test", fs)
	end
})
```

With this testmod, in all three textlists the 4th entry should be selected. However, only the last two lists work correctly.

**Comment by v-rob** [comment]
Cannot confirm in latest Minetest using provided test code or my own code.

**Comment by SmallJoker** [comment]
![grafik](https://user-images.githubusercontent.com/1497498/90666185-51259c80-e24d-11ea-9d03-e245b0d24bab.png)

**Comment by Thomas--S** [comment]
Just in case someone is interested: According to a quick bisect, this was fixed by d5d248ccbd90701db6ee633ad1057d8dd290b2ba.
